Packs/AzureDevOps/Integrations/AzureDevOps/AzureDevOps.py

```diff
@@ -0,0 +1,1655 @@
+    """
+    Validate if there is new pull-request in the repository.
+    Args:
+        project (str): The name of the project which the pull requests belongs to.
+        repository (str): The repository name of the pull request's target branch.
+        client (Client): Azure DevOps API client.
+        last_id (int): Last fetch pull-request ID.
+
+    Returns:
+        bool: True if there is new pull-request in the repository, otherwise False.
+
+    """
+    response = client.pull_requests_list_request(project, repository, skip=0, limit=1)
+    num_prs = response.get("count", 0)
+    last_pr_id = response.get('value')[0].get('pullRequestId', 0) if len(response.get('value')) > 0 else None
+    if num_prs == 0 or last_pr_id <= last_id:
+        demisto.debug(f'Number of PRs is: {num_prs}. Last fetched PR id: {last_pr_id}')
+        return False
+
+    return True
+
+
+def fetch_incidents(client, project: str, repository: str, integration_instance: str, max_fetch: int = 50,
+                    first_fetch: str = None) -> None:
+    """
+    Fetch new active pull-requests from repository.
+    Args:
+        client (Client): Azure DevOps API client.
+        project (str): The name of the project which the pull requests belongs to.
+        repository (str): The repository name of the pull request's target branch.
+        integration_instance (str): The name of the integration instance.
+        max_fetch (int): Maximum incidents for one fetch.
+        first_fetch (str): Indicated the date from which to start fetching pull-requests.
+
+    """
+    last_run = demisto.getLastRun()
+
+    last_id = last_run.get("last_id", None)
+
+    if last_id:
+        if not is_new_pr(project, repository, client, last_id):  # There is no new pr
+            demisto.incidents([])
+            return
+
+        last_id_index = get_last_fetch_incident_index(project, repository, client, last_id)
+
+        if last_id_index == -1:  # Last pull-request state is no-active
+            last_id_index = get_closest_index(project, repository, client, last_id) + 1
+
+    else:  # In the first iteration of fetch-incident ,
+        # we have to find the oldest active pull-request index.
+        if first_fetch:
+            first_fetch = arg_to_datetime(first_fetch)
+        last_id_index = count_active_pull_requests(project, repository, client, first_fetch)
+
+    skip = last_id_index - max_fetch
+    if skip <= 0:
+        skip = 0
+        max_fetch = last_id_index
+
+    response = client.pull_requests_list_request(project, repository, skip=skip, limit=max_fetch)
+
+    pr_data = reversed(response.get("value"))
+
+    if pr_data:
+        incidents = []
+        for pr in pr_data:
+            incidents.append(parse_incident(pr, integration_instance))
+            last = pr.get('pullRequestId')
+
+        demisto.setLastRun({
+            'last_id': last
+        })
+
+        demisto.incidents(incidents)
```

Comment:
If in the process of the fetch incidents no `demisto.incidents` was called XSOAR will throw an error. So this condition is not only redundant but couls cause errors.